src/features/courses/course-controller.ts

```diff
@@ -2717,62 +2722,90 @@ class CourseController {
     }
 
     async createGradeInstancesForAssessment(options: CreateGradeInstancesForAssessmentOptions): Promise<StudentTopicAssessmentInfo> {
-        const { topicId, userId } = options;
-        const topic = await courseRepository.getCourseTopic({id: topicId});
-        const topicInfo = await topic.getTopicAssessmentInfo(); // order by startDate 
-
-        const questions = await courseRepository.getQuestionsFromTopicId({id:topicId});
-        const startTime = moment().add(1,'minute'); // 1 minute should cover any delay in creating records before a student can actually begin
-
-        let endTime = moment(startTime).add(topicInfo.duration, 'minutes');
-        if (topicInfo.hardCutoff) {
-            endTime = moment.min(topic.endDate.toMoment(), endTime);
-        } 
-
-        const nextVersionAvailableTime = moment(startTime).add(topicInfo.versionDelay, 'minutes');
-        // in trying to be clever about shuffling the order, don't forget to shift from 0..n-1 to 1..n
-        const problemOrder = (topicInfo.randomizeOrder) ? _.shuffle([...Array(10).keys()]) : [...Array(10).keys()];
-
-        const studentTopicAssessmentInfo = await this.createStudentTopicAssessmentInfo({
-            userId,
-            topicAssessmentInfoId: topicInfo.id,
-            startTime: startTime.toDate(),
-            endTime: endTime.toDate(),
-            nextVersionAvailableTime: nextVersionAvailableTime.toDate(),
-            maxAttempts: topicInfo.maxGradedAttemptsPerVersion,
-        });
+        return useDatabaseTransaction(async (): Promise<StudentTopicAssessmentInfo> => {
+            const { topicId, userId } = options;
+            const topic = await courseRepository.getCourseTopic({id: topicId});
+            const topicInfo = await topic.getTopicAssessmentInfo(); // order by startDate 
+    
+            const questions = await courseRepository.getQuestionsFromTopicId({id:topicId});
+            const startTime = moment().add(1,'minute'); // 1 minute should cover any delay in creating records before a student can actually begin
+    
+            let endTime = moment(startTime).add(topicInfo.duration, 'minutes');
+            if (topicInfo.hardCutoff) {
+                endTime = moment.min(topic.endDate.toMoment(), endTime);
+            } 
+    
+            const nextVersionAvailableTime = moment(startTime).add(topicInfo.versionDelay, 'minutes');
+            // in trying to be clever about shuffling the order, don't forget to shift from 0..n-1 to 1..n
+            const problemOrder = (topicInfo.randomizeOrder) ? _.shuffle([...Array(10).keys()]) : [...Array(10).keys()];
```

Comment:
Why is this hardcoded to 10?

Reply:
This isn't actually one of my changes, this is already in dev, if you check without whitespace this was just the transaction fix

Reply:
This might be @doombot-exe 's code